wagon.py

```diff
@@ -428,7 +429,10 @@ def _get_os_properties():
     and will be removed in Python 3.7. By that time, distro will become
     mandatory.
     """
-    return linux_distribution(full_distribution_name=False)
+    try:
+        return linux_distribution(full_distribution_name=False)
+    except TypeError:
+        return None, None, None
```

Comment:
maybe return the `{'distribution': xxx if x else None}` dict from here already, because this function is only ever called in order to construct this dict later